tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -635,4 +635,50 @@ function() {
 		$tag = wp_get_attachment_image( $attachment_id, 'medium', false, array( 'class' => "wp-image-{$attachment_id}" ) );
 		$this->assertNotSame( $tag, webp_uploads_img_tag_update_mime_type( $tag, 'the_content', $attachment_id ) );
 	}
+
+	/**
+	 * Tests whether additional mime types generated only for allowed image sizes or not when the filter is used.
+	 *
+	 * @test
+	 */
+	public function it_should_create_mime_types_for_allowed_sizes_only_via_filter() {
+		add_filter(
+			'webp_uploads_image_sizes_with_additional_mime_type_support',
+			function( $sizes ) {
+				$sizes['allowed_size_400x300'] = true;
+				return $sizes;
+			}
+		);
+
+		add_image_size( 'allowed_size_400x300', 400, 300, true );
+		add_image_size( 'not_allowed_size_200x150', 200, 150, true );
+
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/car.jpeg' );
+
+		remove_image_size( 'allowed_size_400x300' );
+		remove_image_size( 'not_allowed_size_200x150' );
+
+		$this->assertImageHasSizeSource( $attachment_id, 'allowed_size_400x300', 'image/webp' );
+		$this->assertImageNotHasSizeSource( $attachment_id, 'not_allowed_size_200x150', 'image/webp' );
+	}
+
+	/**
+	 * Tests whether additional mime types generated only for allowed image sizes or not when the global variable is updated.
+	 *
+	 * @test
+	 */
+	public function it_should_create_mime_types_for_allowed_sizes_only_via_global_variable() {
+		add_image_size( 'allowed_size_400x300', 400, 300, true );
+		add_image_size( 'not_allowed_size_200x150', 200, 150, true );
+
+		$GLOBALS['_wp_additional_image_sizes']['allowed_size_400x300']['provide_additional_mime_types'] = true;
```

Comment:
I assume this could be removed once we update `add_image_size` to accept the new "output_additional_mimes" parameter, right? Can you add a TODO comment to remove this once that is in place?